### DISCUSSION TITLE
Question on relay internal functionality features and potential bug report.

### DISCUSSION THREAD

**Discussion by McFow1er** [FakeFishGames/Barotrauma#5031] [Code questions and modding support]
Okay so first off I'd like to know if the relays limiting current flow to the load instead of letting full or relative partial supply through even with head room from the max current limiter is an intended feature or not and if this functionality were to be fixed in a pull request would it likely be added.

Either way of it being a bug or feature here's what I've observed so far and wanted to address:
I've discovered what very well may be the reason relays in series are so buggy and cause power flickering. When being used normally relays will function as expected, they will flow as much power as the set maximum will allow to supply load, however trying to overload the relay will result in it only flowing the amount needed to supply the load until that supply value is reached in which it will then immediately spike up to the current supply wattage for a few seconds before tapering off back down to the current load then hitting that value and shooting back up to the current supply, constantly repeating this process of full supply and dropping down to load level.

The reason for asking if this functionality is intentional or not is for the sake of being able to overload relays themselves with the appropriate options enabled on the item and items later on down the chain when used in power circuits, for example junction boxes as a main target or on various devices that have overvolt damage enabled. Though given the fact that items connected to the relay are essentially placed onto a new power grid is the reason for power limiting being a thing on relays so that new grid doesn't receive overload damage when the rest of the grid should be fine.

**Comment by NilanthAnimosus** [comment]
Im no expect on power or relays but they still seem odd even in the current version for some uses.

the power limit doesn't get exceeded from the sounds of your testing with it, may be a feature as you could set the max to 5000 then - but still have a high voltage as voltage is based on current flow vs power demand (twice the power in than needed is a voltage of 2.0) - so you can still overload with a power cap.

Now just to explain one example of unusual behaviour with these - on my own customized submarine not on the workshop I tried to make all emergency lights require power, as well as all lights - then tie it all into the wiring grid using a mix of relays and junctions.

It simply was 1 junction wired to 4 relays at a time, but the result I had was every single relay operating at 50% load when the junction was supplied via battery (Capable of outputting at full). relays have always been slightly dodgy and a bit of an enigma, 

I would see four relays at 50 power used each, with say 100 used and 200 wanted always having my voltage at 50% or 25% (Its hard to recall now, I 'fixed' this by making all my lights run from 0.0 min voltage and raised the light values). the overall power logic I have always found is a hard to comprehend spiderweb of code on a good day though.

I think in the past relays used to explode for no reason at all - and were changed to not straight up overload, perhaps they break once again though in these newer versions I typically out of habit disable overloading if they have it for relays specifically. I think relays likely have several unintended strange effects still, they may be capable of power duplication for instance. they act like separate individual networks if I recall which is how the behavior is strange to begin with due to one-way power transfer into that separate network.

The reason I bring all the above up is if they dont overload currently, its probably because their behaviour still doesn't function quite as you would expect and the power math is off in a lot of edge cases that would break many submarines. and would likely expload where on paper it seems completely fine with math in your head. the relays have some quirks that I could see this being a temporary patch to stop problems on a very old, different issue.

I do think these need looking into, and maybe some information on their current state though so im interested in this as well.
